File: backend/src/main/java/csc13001/plantpos/domain/models/Inventory.java
package csc13001.plantpos.domain.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@Entity
@Table(name = "inventories")
public class Inventory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long batchId;

    @NotBlank(message = "Supplier is required")
    @Column(nullable = false, length = 255)
    private String supplier;

    @NotNull(message = "Product is required")
    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @NotNull(message = "Quantity is required")
    @Min(value = 1, message = "Quantity must be greater than 0")
    @Column(name = "quantity", nullable = false)
    private int quantity;

    @NotNull(message = "Purchase price is required")
    @Min(value = 0, message = "Purchase price must be greater than 0")
    @Column(name = "purchase_price", nullable = false, precision = 18, scale = 2)
    private BigDecimal purchasePrice;

    @Column(name = "purchase_date", nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date purchaseDate;

    public Inventory() {
        this.purchaseDate = new Date();
    }

    public void decreaseStock(int quantity) {
        this.quantity -= quantity;
    }
}

File: csc13001-plant-pos-frontend.Core/Models/Inventory.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace csc13001_plant_pos_frontend.Core.Models;

[Table("inventory")]
public class Inventory
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int BatchId
    {
        get; set;
    }

    [Required]
    [MaxLength(255)]
    public string Supplier { get; set; } = string.Empty;

    [ForeignKey("product")]
    public int ProductId
    {
        get; set;
    }

    public int Quantity
    {
        get; set;
    }

    [Column(TypeName = "decimal(18,2)")]
    public decimal PurchasePrice
    {
        get; set;
    }

    [Column(TypeName = "datetime")]
    public DateTime PurchaseDate { get; set; } = DateTime.Now;

    public Product Product
    {
        get; set;
    }
}
